storage/range.go

```diff
@@ -645,14 +748,21 @@ func (r *Range) processRaftCommand(idKey cmdIDKey, index uint64,
 	return err
 }
 
-// startGossip periodically gossips the cluster ID if it's the
-// first range and the raft leader.
-func (r *Range) startGossip(stopper *util.Stopper) {
+// startGossip periodically gossips the cluster ID if it's the first
+// range. This function should only be called if this range replica is
+// the raft leader.
+func (r *Range) startGossip(term uint64, stopper *util.Stopper) {
 	stopper.RunWorker(func() {
+		r.maybeGossipClusterID()
+		r.maybeGossipFirstRange()
+
 		ticker := time.NewTicker(ttlClusterIDGossip / 2)
 		for {
 			select {
 			case <-ticker.C:
+				if held, curTerm := r.HasLeaderLease(); !held || curTerm != term {
```

Comment:
We always want to gossip the first range even when there is no write pressure. That means we either need to test for raft leadership instead of the lease here, or ensure that we always maintain a lease for the first range as a special case even when there is no write pressure.